storagenode/console/consoleserver/server.go

```diff
@@ -127,163 +105,104 @@ func (server *Server) Close() error {
+func (server *Server) satellitesHandler(wr http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	defer mon.Task()(&ctx)(nil)
 
-	space, err := server.getStorage(ctx, satelliteID)
-	if err != nil {
-		return response, err
+	if req.Method != http.MethodGet {
+		http.Error(wr, "method not allowed", http.StatusMethodNotAllowed)
+		return
 	}
 
-	usage, err := server.getBandwidth(ctx, satelliteID)
+	data, err := server.service.GetAllSatellitesData(ctx)
 	if err != nil {
-		return response, err
+		server.writeError(wr, http.StatusInternalServerError, err)
+		return
 	}
 
-	walletAddress := server.service.GetWalletAddress(ctx)
+	server.writeData(wr, data)
+}
 
-	versionInfo := server.service.GetVersion(ctx)
+// satelliteHandler handles satellite api requests
+func (server *Server) satelliteHandler(wr http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	defer mon.Task()(&ctx)(nil)
 
-	err = server.service.CheckVersion(ctx)
-	if err != nil {
-		return response, err
+	if req.Method != http.MethodGet {
+		http.Error(wr, "method not allowed", http.StatusMethodNotAllowed)
+		return
 	}
 
-	bandwidthChartData, err := server.getBandwidthChartData(ctx, satelliteID)
+	satelliteID, err := storj.NodeIDFromString(strings.TrimLeft(req.URL.Path, "/api/satellite/"))
 	if err != nil {
-		return response, err
+		server.writeError(wr, http.StatusBadRequest, err)
+		return
 	}
 
-	// TODO: uncomment in future, when caching will be implemented
-	// diskSpaceChartData, err := server.getDiskSpaceChartData(ctx, satelliteID, satellites)
-	// if err != nil {
-	// 	return response, err
-	// }
-
-	uptime := server.service.GetUptime(ctx)
-	nodeID := server.service.GetNodeID(ctx)
-
-	// TODO: uncomment in future, when caching will be implemented
-	// if satelliteID != nil {
-	// 	satelliteStats, err := server.service.GetStatsFromSatellite(ctx, *satelliteID)
-	// 	if err != nil {
-	// 		return response, err
-	// 	}
-	//
-	// 	response.UptimeCheck = satelliteStats.UptimeCheck
-	// 	response.AuditCheck = satelliteStats.AuditCheck
-	// }
-
-	response.DiskSpace = *space
-	response.Bandwidth = *usage
-	response.WalletAddress = walletAddress
-	response.VersionInfo = versionInfo
-	response.IsLastVersion = true
-	response.Uptime = uptime
-	response.NodeID = nodeID.String()
-	response.Satellites = satellites
-	response.BandwidthChartData = bandwidthChartData
-	//response.DiskSpaceChartData = diskSpaceChartData
-
-	return response, nil
-}
+	if err = server.service.VerifySatelliteID(ctx, satelliteID); err != nil {
+		server.writeError(wr, http.StatusNotFound, err)
+		return
+	}
 
-func (server *Server) getBandwidth(ctx context.Context, satelliteID *storj.NodeID) (_ *console.BandwidthInfo, err error) {
-	if satelliteID != nil {
-		return server.service.GetBandwidthBySatellite(ctx, *satelliteID)
+	data, err := server.service.GetSatelliteData(ctx, satelliteID)
+	if err != nil {
+		server.writeError(wr, http.StatusInternalServerError, err)
```

Comment:
Doesn't error has a stack trace?

Reply:
Yes, it does but the stack trace is generated when the error is created by zeebo/errs package and now I see that it will be enough to find out that it's coming from the specific handler function, yesterday I thought that it won't, hence options 2 should be preferred.

Reply:
But do we need to log when client requested untrusted satellite or provide bad satellite name that can't be parsed. Also that is storagenode private server, and the only client is most likely the storagenode owner himself.

Reply:
Anyway, moved it to `server.writeError`

Reply:
I wouldn't log any error which we know the reason.
For example, I wouldn't log `bad satellite name that can't be parsed` the server is acting well if it detects that's bad and it rejects the client because it is sending bad data, so I wouldn't want to see that error in the logs from the server point of view.

The problem with `internal errors` is that we don't know what could be wrong and all the information as possible is welcome to figure it out; ultimately if we have many of them in a short period of time we are sure that something is not behaving well and it's leading the system to a breakdown.